tests/test_wrappers.py

```diff
@@ -0,0 +1,148 @@
+        self.assertEqual(repr(ilist), "ImmutableList(['Ahem', 'White Space', 'serif'])")
+        self.assertEqual(len(ilist), 3)
+
+        # Check get item
+        self.assertEqual(ilist[0], 'Ahem')
+        self.assertEqual(ilist[-1], 'serif')
+
+        # Check immutable
+        with self.assertRaises(TypeError):
+            ilist[0] = 'initial'
+
+        # Check equality
+        ilist1 = ImmutableList(['Ahem', 2])
+        ilist2 = ImmutableList(['Ahem', 2])
+        ilist3 = ImmutableList([2, 'Ahem'])
+        self.assertEqual(ilist1, ilist2)
+        self.assertNotEqual(ilist1, ilist3)
+
+        # Check hash
+        self.assertEqual(hash(ilist1), hash(ilist2))
+
+        # Check copy
+        self.assertNotEqual(id(ilist1), id(ilist2))
+        self.assertNotEqual(id(ilist1), id(ilist1.copy()))
+        self.assertNotEqual(id(ilist2), id(ilist1.copy()))
+        self.assertEqual(hash(ilist2), hash(ilist1.copy()))
+        self.assertEqual(ilist1, ilist1.copy())
+
+
+class FontFamilyTests(TestCase):
+
+    def test_fontfamily(self):
+        # Check initial
+        font = FontFamily()
+        self.assertEqual(str(font), '')
+        self.assertEqual(repr(font), 'FontFamily()')
+        self.assertEqual(len(font), 0)
+
+        # Check value
+        font = FontFamily(['Ahem'])
+        self.assertEqual(str(font), 'Ahem')
+        self.assertEqual(repr(font), "FontFamily(['Ahem'])")
+        self.assertEqual(len(font), 1)
+
+        # Check values
+        font = FontFamily(['Ahem', 'White Space', 'serif'])
+        self.assertEqual(str(font), 'Ahem, "White Space", serif')
+        self.assertEqual(repr(font), "FontFamily(['Ahem', 'White Space', 'serif'])")
+        self.assertEqual(len(font), 3)
+
+        # Check get item
+        self.assertEqual(font[0], 'Ahem')
+        self.assertEqual(font[-1], 'serif')
+
+        # Check immutable
+        with self.assertRaises(TypeError):
+            font[0] = 'initial'
+
+        # Check equality
+        font1 = FontFamily(['Ahem', 'serif'])
+        font2 = FontFamily(['Ahem', 'serif'])
+        font3 = FontFamily(['serif', 'Ahem'])
+        self.assertEqual(font1, font2)
+        self.assertNotEqual(font1, font3)
+
+        # Check hash
+        self.assertEqual(hash(font1), hash(font2))
+
+        # Check copy
+        self.assertNotEqual(id(font1), id(font2))
+        self.assertNotEqual(id(font1), id(font1.copy()))
+        self.assertNotEqual(id(font2), id(font2.copy()))
+        self.assertEqual(font1, font1.copy())
+
+
+class ShorthandTests(TestCase):
+
+    def test_shorthand(self):
+        shorthand = Shorthand()
+        self.assertEqual(str(shorthand), 'Shorthand()')
+        self.assertEqual(repr(shorthand), ("Shorthand()"))
+
+
+class FontShorthandTests(TestCase):
+
+    def test_font_shorthand(self):
+        # Check initial
+        font = FontShorthand()
+        self.assertEqual(str(font), 'normal normal normal medium/normal initial')
```

Comment:
Fair enough :-)

Reply:
@freakboy3742 I kinda disagree because setting the font shorthand to say just `'initial'`, will reset all other font properties to their default values so this is like effectively setting those values to their initial values "implicitly"

I am not sure how we could really handle this use case.

I could try to display the minimum values in this str but once this "Font" object is applied to the style of a given node it is effectively resetting all the other properties.

Reply:
Still need to think this a bit further